packages/serverpod/lib/src/database/adapters/postgres/database_connection.dart

```diff
@@ -378,38 +401,17 @@ class DatabaseConnection {
     String query, {
     int? timeoutInSeconds,
     Transaction? transaction,
+    QueryMode? queryMode,
   }) async {
-    var postgresTransaction = _castToPostgresTransaction(transaction);
-
-    var startTime = DateTime.now();
-    try {
-      var context =
-          postgresTransaction?.executionContext ?? _postgresConnection;
+    var result = await _query(
```

Comment:
Question: if we are passing everything to query, do we need to have the execute method still? 

What is the real difference here between execute and query?

Reply:
The real different is the return value, execute return the number of rows effected while query returns a result.